File: check_dates.py
import csv
def check_dates(file_path, col_names):
    print(f"--- {file_path} ---")
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            reader = csv.DictReader(f)
            unique_vals = {col: set() for col in col_names}
            row_count = 0
            for row in reader:
                row_count += 1
                for col in col_names:
                    if col in row and row[col]:
                        val = row[col].strip()
                        if val:
                            unique_vals[col].add(val)
                            
            for col in col_names:
                vals = list(unique_vals[col])
                print(f"{col} - Total unique: {len(vals)}")
                print(f"Samples: {vals[:15]}")
            print(f"Total Rows: {row_count}")
    except Exception as e:
        print(f"Error: {e}")

check_dates('/mnt/d/OneDrive/Desktop/samuel-haddad/holy-squat/sources/sessions_bkup.csv', ['date'])
check_dates('/mnt/d/OneDrive/Desktop/samuel-haddad/holy-squat/sources/workouts_bkup.csv', ['date', 'workout_date'])
check_dates('/mnt/d/OneDrive/Desktop/samuel-haddad/holy-squat/sources/workouts_bkup_logs.csv', ['workout_date'])

File: check_dates.dart
import 'dart:io';

void checkDates(String filePath, List<String> colNames) {
  print('--- \$filePath ---');
  try {
    final file = File(filePath);
    final lines = file.readAsLinesSync();
    if (lines.isEmpty) return;
    
    final headerStr = lines.first;
    // Simple split for header
    final headers = headerStr.split(RegExp(r',(?=(?:[^"]*"[^"]*")*[^"]*\$)')).map((e) => e.trim().replaceAll('"', '')).toList();
    
    final Map<String, int> colIndices = {};
    for (var col in colNames) {
      final idx = headers.indexOf(col);
      if (idx != -1) {
        colIndices[col] = idx;
      }
    }
    
    final Map<String, Set<String>> uniqueVals = { for (var item in colNames) item : <String>{} };
    
    for (int i = 1; i < lines.length; i++) {
      final line = lines[i];
      if (line.trim().isEmpty) continue;
      
      final parts = line.split(RegExp(r',(?=(?:[^"]*"[^"]*")*[^"]*\$)')).map((e) => e.trim().replaceAll('"', '')).toList();
      for (var col in colNames) {
        if (colIndices.containsKey(col)) {
          final idx = colIndices[col]!;
          if (idx < parts.length) {
            final val = parts[idx];
            if (val.isNotEmpty) {
              uniqueVals[col]!.add(val);
            }
          }
        }
      }
    }
    
    for (var col in colNames) {
      final vals = uniqueVals[col]!.toList();
      print('\$col - Total unique: \${vals.length}');
      print('Samples: \${vals.take(15).toList()}');
    }
    print('Total Rows: \${lines.length - 1}');
  } catch (e) {
    print('Error: \$e');
  }
}

void main() {
  checkDates(r'D:\OneDrive\Desktop\samuel-haddad\holy-squat\sources\sessions_bkup.csv', ['date']);
  checkDates(r'D:\OneDrive\Desktop\samuel-haddad\holy-squat\sources\workouts_bkup.csv', ['date', 'workout_date']);
  checkDates(r'D:\OneDrive\Desktop\samuel-haddad\holy-squat\sources\workouts_bkup_logs.csv', ['workout_date']);
}
